File: src/main/java/com/runix/parser/Token.java
// src/main/java/com/runix/parser/Token.java
package com.runix.parser;

public class Token {
    public enum Type {
        IDENTIFIER, NUMBER, STRING, OPERATOR, EOF
    }
    public final Type type;
    public final String value;
    public Token(Type type, String value) {
        this.type = type;
        this.value = value;
    }
    @Override
    public String toString() {
        return "[" + type + ": \"" + value + "\"]";
    }
}

File: go/parser/token.go
package parser

type TokenType int

const (
	IDENTIFIER TokenType = iota
	NUMBER
	STRING
	OPERATOR
	EOF
)

type Token struct {
	Type  TokenType
	Value string
}

func (t Token) String() string {
	return "[" + t.Type.String() + ": " + t.Value + "]"
}

func (tt TokenType) String() string {
	switch tt {
	case IDENTIFIER:
		return "IDENTIFIER"
	case NUMBER:
		return "NUMBER"
	case STRING:
		return "STRING"
	case OPERATOR:
		return "OPERATOR"
	case EOF:
		return "EOF"
	default:
		return "UNKNOWN"
	}
}
